apps/andi/test/unit/andi_web/controllers/api/ingestion_controller_test.exs

```diff
@@ -162,13 +172,57 @@ defmodule AndiWeb.API.IngestionControllerTest do
 
   describe "PUT /ingestion" do
     test "PUT /api/ with data returns a 201", %{conn: conn} do
+      smrt_ingestion = TDG.create_ingestion(%{})
+      {_, ingestion_without_id} = smrt_ingestion |> struct_to_map_with_string_keys() |> Map.pop("id")
+
+      allow(Brook.Event.send(@instance_name, any(), any(), any()), return: :ok)
+      allow(Andi.InputSchemas.Datasets.get(any()), return: %{technical: %{sourceType: "ingest"}})
+
+      conn = put(conn, @route, ingestion_without_id)
+
+      {_, decoded_body} = Jason.decode(response(conn, 201))
+
+      expected_ingestion = TDG.create_ingestion(Map.put(ingestion_without_id, "id", decoded_body["id"]))
```

Comment:
Do you need the TDG.create_ingestion? The Map.put should give you the full ingestion already.

Reply:
Yeah, but that's a map. It's expecting an Ingestion ... object? Not sure what the elixir term is for it.